includes/data-port/class-sensei-data-port-job.php

```diff
@@ -191,16 +200,34 @@ public function get_results() {
 	 * @return array The logs.
 	 */
 	public function get_logs() {
-		return array_map(
-			function ( $log ) {
-				return [
-					'message' => $log[0],
-					'level'   => $log[1],
-					'data'    => $log[2],
-				];
-			},
-			$this->logs
-		);
+		$ordered_logs = [];
+
+		foreach (
+			[
+				'__generic',
+				Sensei_Import_Course_Model::MODEL_KEY,
```

Comment:
This is importer logic in the generic class. It also doesn't sort by line number (can be mixed up because or prerequisite handling; per the attached issue). It might be better to just do a `usort` with a function that sorts first by data `type` (if it exists, if not, have it be sorted at top) and then `line` within a type.

Reply:
Totally missed the case where line sorting would be messed up. Should be ok in the new implementation: 386bfcd1673efbec233af6bbd5b324860df1ff2b